### PR TITLE
planner: fix the issue that MVIndex leads to "can't find plan" error

### LINKED ISSUE #49438: query reports "Can't find a proper physical plan for this query"
## Bug Report

Please answer these questions before submitting your issue. Thanks!

### 1. Minimal reproduce step (Required)

``` SQL
CREATE TABLE `tl1b9523cf` (
  `col_1` double DEFAULT NULL,
  `col_2` date DEFAULT '2032-02-22',
  `col_3` varbinary(337) NOT NULL DEFAULT 'jeHvC6^7v',
  `col_4` varbinary(247) NOT NULL,
  `col_5` timestamp NOT NULL,
  `col_6` json DEFAULT NULL,
  `col_7` json NOT NULL,
  `col_8` enum('Alice','Bob','Charlie','David') COLLATE utf8_unicode_ci NOT NULL DEFAULT 'David',
  KEY `idx_1` (`col_2`,`col_1`),
  KEY `idx_2` (`col_1`),
  KEY `idx_3` (`col_3`(1),`col_8`,(cast(`col_6` as datetime array))),
  UNIQUE KEY `idx_4` (`col_1`,(cast(`col_7` as date array)),`col_3`)
) ENGINE=InnoDB DEFAULT CHARSET=utf8 COLLATE=utf8_unicode_ci;
desc select  tl1b9523cf.col_3 as r0 , tl1b9523cf.col_1 as r1 from tl1b9523cf where tl1b9523cf.col_1 in ( 5286.70512761001 ,1720.6188106198429 ,6342.041981907588 ,4051.931997825317 );
```

### 2. What did you expect to see? (Required)
No error
### 3. What did you see instead (Required)
``` SQL
mysql> desc select  tl1b9523cf.col_3 as r0 , tl1b9523cf.col_1 as r1 from tl1b9523cf where tl1b9523cf.col_1 in ( 5286.70512761001 ,1720.6188106198429 ,6342.041981907588 ,4051.931997825317 );
ERROR 1815 (HY000): Internal : Can't find a proper physical plan for this query
```
### 4. What is your TiDB version? (Required)
nightly-a7260ff598ce0e7c0a77367e84ec265f91b8073e

### PR DESCRIPTION
**Pull request by qw4990**
### What problem does this PR solve?

Issue Number: close #49438

Problem Summary:

### Check List

Tests 

- [x] Unit test
- [ ] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [ ] No need to test
  > - [ ] I checked and no code files have been changed.
  > 

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

Please refer to [Release Notes Language Style Guide](https://pingcap.github.io/tidb-dev-guide/contribute-to-tidb/release-notes-style-guide.html) to write a quality release note.

```release-note
None
```

### CODE REVIEW

File: pkg/planner/core/casetest/integration_test.go [hunk 1]
```diff
@@ -421,6 +421,14 @@ func TestFixControl45132(t *testing.T) {
 	tk.MustHavePlan(`select * from t where a=2`, `TableFullScan`)
 }
 
+func TestIssue49438(t *testing.T) {
+	store := testkit.CreateMockStore(t)
+	tk := testkit.NewTestKit(t, store)
+	tk.MustExec(`use test`)
+	tk.MustExec(`create table tx (a int, b json, key k(a, (cast(b as date array))))`)
```
Comment (AilinKid) [inline pkg/planner/core/casetest/integration_test.go:428]: better add `drop table if exists tx`

**Review by AilinKid — APPROVED** [review]

File: pkg/planner/core/casetest/integration_test.go (hunk 1 above)
Comment (qw4990) [reply pkg/planner/core/casetest/integration_test.go:429]: updated

**Review by time-and-fate — APPROVED** [review]

**Pull request merged**